### PR TITLE
Lesson Statements

### PR DESCRIPTION
**Pull request by cliffcaseyyet**
- content page viewed
- essay assessed
- lesson ended
- lesson restarted
- lesson resumed
- lesson started
- question answered (all cmi subtypes)
- question viewed (all cmi subtypes)

Along the way:
- Added result to module completion
- Added a few correctresponse options to cmi subtypes

### CODE REVIEW

File: tests/mod_lesson/content_page_viewed/statements.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          "http://moodle.org": "1.0.0",
+          "https://github.com/xAPI-vle/moodle-logstore_xapi": "0.0.0-development",
+          "event_name": "\\mod_lesson\\event\\content_page_viewed",
+          "event_function": "\\src\\transformer\\events\\mod_lesson\\content_page_viewed"
+        }
+      },
+      "contextActivities": {
+        "parent": [
+          {
+            "id": "http://www.example.org/mod/lesson/view.php?id=1",
```
Comment (milt) [inline tests/mod_lesson/content_page_viewed/statements.json:38]: why don't I see the hash here?

Comment (milt) [reply tests/mod_lesson/content_page_viewed/statements.json:38]: oh nm this is the course module sorry

File: tests/mod_lesson/question_answered/choice/statements.json [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+              "name": {
+                "en": "test_name"
+              }
+            }
+          }
+        ]
+      }
+    }
+  }
+]
```
Comment (milt) [inline tests/mod_lesson/question_answered/choice/statements.json]: github is giving me a no newline warning on this one

**Review by milt — CHANGES_REQUESTED** [review]
As discussed on slack, come up with a type for the "inner" lesson instance and add it to the profile.

File: tests/mod_lesson/question_answered/choice/statements.json (hunk 2 above)
Comment (cliffcaseyyet) [reply tests/mod_lesson/question_answered/choice/statements.json]: 9275d62318f9da25310494c116876813fa36cd6d

**Comment by cliffcaseyyet** [comment]
> As discussed on slack, come up with a type for the "inner" lesson instance and add it to the profile.

2e8bd1b5bbc86e14e8a2ed50966f0d24146eb574 <- that is the code, moodle-sample-statements is already updated. But what is your process for the profile?

**Review by milt — APPROVED** [review]

**Pull request merged**